R/model_based_starting_dose.R

```diff
@@ -73,8 +69,31 @@ model_based_starting_dose <- function(
 
   ## Call core function
   starting_dose <- do.call(dose_grid_search, args)
-
-  ## set all amounts in regimen to selected dose
-  reg$dose_amts <- rep(starting_dose, length(reg$dose_amts))
+  
+  ## elongate regimen, if multiple time-varying targets specified
```

Comment:
maybe this could use a little more explanation. i think i understand why we need to do this, but i'm worried about possible interactions with regimens like busulfan, where we never want to change the number of doses

Reply:
Added more context in comment